cachi2/core/package_managers/pip.py

```diff
@@ -1424,6 +1424,9 @@ def _download_dependencies(
             if allow_binary:
                 to_download.extend(w for w in wheels if not w.path.exists())
 
+            if source is None:
+                continue
```

Comment:
Just remembered that this won't be that simple because of https://github.com/containerbuildsystem/cachi2/pull/255#discussion_r1305581494

We must report the downloaded dependency if we download at least 1 archive for it, we can't omit it just because it doesn't have sdists